### PR TITLE
7) Fix getTables request

### LINKED ISSUE #72: Failed to get tables: 'Undefined index: INFORMATION_SCHEMA.COLUMNS
I am trying to download account usage https://connection.keboola.com/admin/projects/258/extractors/keboola.ex-db-snowflake/461070099

https://connection.keboola.com/admin/utils/logs?file=2018/11/07/10/2018-11-07-10-26-10-5be2afb2bbfe6-exception

In the UI it works:
https://connection.keboola.com/admin/projects/258/extractors/keboola.ex-db-snowflake/461070099

### PR DESCRIPTION
**Pull request by pivnicek**
Fixes #72 

Rebased onto #71

### CODE REVIEW

**Comment by Halama** [comment]
I've rebased it

File: tests/SnowflakeTest.php [hunk 1]
```diff
@@ -222,7 +222,7 @@ public function testRunEmptyQuery(): void
         $this->assertFileNotExists($outputManifestFile);
     }
 
-    public function testGetTables(): void
+    public function testGetTablesWithSchema(): void
```
[the remark this answers is not in the document]
Comment (Halama) [reply tests/SnowflakeTest.php]: I was able to manually `php run.php  --data=/code/data/` confirm that this change is fixing it.
`data/config.yml`
```
action: getTables
parameters:
    data_dir: /code/tests/data
    db: { user: martin_snflk_extractor, '#password': qHtXQTDvZn7D6c6nK, host: kebooladev.snowflakecomputing.com, port: '443', database: martin_snflk_extractor, schema: snowflake_extractor, warehouse: DEV, password: *** }
    extractor_class: Snowflake
    tables: []
```

`create table columns ("id" varchar, "name" varchar);`

**Comment by pivnicek** [comment]
@Halama , sorry I let this slip, I've updated it and think it's now more clear.

**Comment by pivnicek** [comment]
@Halama az budes mit kvilicku muzes na tohle dat oko?  rad bych ho releasnout.

**Comment by Halama** [comment]
@Halama I am not sure when I'll get to this:( Maybe try someone other for review?

**Comment by pivnicek** [comment]
@Halama sure, no worries

File: composer.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
         "keboola/php-csv-db-import": "~2.3.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "4.6.*",
+        "phpunit/phpunit": "^7.0",
```
Comment (tomasfejfar) [inline composer.json]: Why the upgrade? And why not upgrade anything else (cs, phpstan)? I have to say I much prefer to create separate PR for any non-related changes.

File: tests/SnowflakeTest.php [hunk 3]
```diff
@@ -522,7 +522,7 @@ public function testGetTablesWithoutSchema(): void
         // add a table to a different schema (should not be fetched)
```
Comment (tomasfejfar) [inline tests/SnowflakeTest.php]: it is fetched though - it's even in the expected data...

File: tests/AbstractSnowflakeTest.php [hunk 4]
```diff
@@ -155,6 +155,8 @@ private function setupTables(): void
                       OBJECT_CONSTRUCT(\'a\', 1, \'b\', \'BBBB\', \'c\', null) AS "org",
                       ARRAY_CONSTRUCT(10, 20, 30) AS "arr";'
         );
+
+        $this->connection->query("DROP TABLE IF EXISTS \"PUBLIC\".\"no_schema_escaping\"");
```
Comment (tomasfejfar) [inline tests/AbstractSnowflakeTest.php]: This is never created, so I think it's enough that it has been ran at least once with this query and now can be safely removed.

File: tests/SnowflakeTest.php (hunk 3 above)
Comment (pivnicek) [reply tests/SnowflakeTest.php]: sorry, the comment is incorrect.  schema is not set in this test so all tables from all schemas should be fetched.  The test name `testGetTablesWithoutSchema` is correct.

File: composer.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (pivnicek) [reply composer.json]: There must have been a reason, I think to match with the common-lib version, but you're right, should have been better done somewhere else.  Shall I move it, or can we keep it just this once ;-)

**Comment by pivnicek** [comment]
Thanks @tomasfejfar !  I've updated it.

**Comment by pivnicek** [comment]
ok, I'm going to merge this then 🎉

**Pull request merged**